packages/lib/next-auth/auth-options.ts

```diff
@@ -23,21 +24,23 @@ export const NEXT_AUTH_OPTIONS: AuthOptions = {
       },
       authorize: async (credentials, _req) => {
         if (!credentials) {
-          return null;
+          throw new Error(ErrorCodes.CredentialsNotFound);
         }
 
         const { email, password } = credentials;
 
-        const user = await getUserByEmail({ email }).catch(() => null);
+        const user = await getUserByEmail({ email }).catch(() => {
+          throw new Error(ErrorCodes.IncorrectEmailPassword);
+        });
 
-        if (!user || !user.password) {
-          return null;
+        if (!user.password) {
+          throw new Error(ErrorCodes.UserMissingPassword);
         }
 
-        const isPasswordsSame = compare(password, user.password);
+        const isPasswordsSame = await compare(password, user.password);
```

Comment:
this seems to be the issue

Reply:
I need to add an eslint rule for this 😓

Reply:
If we have a `no-floating-promises`rule

Reply:
We can combine it with `require-await` rule as well